flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/StateAssignmentOperation.java

```diff
@@ -108,6 +108,7 @@ public void assignStates() {
                 OperatorID operatorID =
                         operatorIDPair
                                 .getUserDefinedOperatorID()
+                                .filter(operatorStates::containsKey)
```

Comment:
this seems not having any effect since `operatorStates` starts with empty.

But why the tests fails after removing this?

Reply:
Sorry this should be a mistake... We should do the check against the `localOperators`. The test current passed due to in this case we always take `generated operator id`, which just coincident with the case we test. I added another case to test another case that should take `uid hash`.